daemons/dss-scalability-test/app.py

```diff
@@ -0,0 +1,254 @@
+def checkout_bundle(event, context):
+    bundle_id = event['bundle']['bundle_id']
+    app.log.info(f"Checkout bundle: {bundle_id}")
+    checkout_output = client.post_bundles_checkout(uuid=bundle_id, replica='aws', email='rkisin@chanzuckerberg.com')
+    return {"job_id": checkout_output['checkout_job_id']}
+
+
+@app.step_function_task(state_name="CheckoutDownloadStatus", state_machine_definition=state_machine_def)
+def checkout_status(event, context):
+    job_id = event['checkout']['job_id']
+    app.log.info(f"Checkout status job_id: {job_id}")
+    # checkout_output = client.get_bundles_checkout(job_id)
+    # return checkout_output['status']
+
+
+@app.step_function_task(state_name="CompleteTest", state_machine_definition=state_machine_def)
+def complete_test(event, context):
+    save_results(event, 'SUCCEEDED')
+
+
+@app.step_function_task(state_name="fallback", state_machine_definition=state_machine_def)
+def fallback(event, context):
+    save_results(event, 'FAILED')
+
+
+def save_results(event, result: str):
+    table = dynamodb.Table('scalability_test')
+    start_time = event['bundle']['start_time']
+
+    expiration_ttl = int(time.time()) + 14 * 24 * 60 * 60   # 14 days
+
+    table.put_item(
+        Item={
+            'run_id': event["test_run_id"],
+            'execution_id': event["execution_id"],
+            'duration': current_time() - start_time - WAIT_CHECKOUT * 1000,
+            'status': result,
+            'created_on': datetime.datetime.now().isoformat(),
+            'expiration_ttl': expiration_ttl
+        }
+    )
+
+
+@app.sns_topic_subscriber("dss-scalability-test-run-" + os.environ["DSS_DEPLOYMENT_STAGE"])
+def launch_test_run(event, context):
+    print('Log test run')
+    msg = json.loads(event["Records"][0]["Sns"]["Message"])
+    run_id = msg["run_id"]
+    table = dynamodb.Table('scalability_test_run')
+    table.put_item(
+        Item={
+            'run_id': run_id,
+            'executions': 0,
+            'succeeded_count': 0,
+            'failed_count': 0,
+            'average_duration': 0,
+            'created_on': datetime.datetime.now().isoformat()
+        }
+    )
+
+@app.sns_topic_subscriber("dss-scalability-test-" + os.environ["DSS_DEPLOYMENT_STAGE"])
+def launch_exec(event, context):
+    msg = json.loads(event["Records"][0]["Sns"]["Message"])
+    run_id = msg["run_id"]
+    execution_id = msg["execution_id"]
+    test_input = {"execution_id": execution_id, "test_run_id": run_id}
+    stepfunctions.step_functions_invoke("dss-scalability-test-{stage}", execution_id, test_input)
+
+@app.dynamodb_stream_handler(table_name="scalability_test", batch_size=50)
+def handle_dynamodb_stream(event, context):
+    success_count = 0
+    failure_count = 0
+    records = 0
+    duration_sum = 0
+    run_id = None
+    for record in event['Records']:
+        if record['eventName'] == 'INSERT':
+            run_id = record['dynamodb']['NewImage']['run_id']['S']
```

Comment:
is `run_id` supposed to change in every loop?

Reply:
no, updated